backend/core/domains/contracts/services.py

```
@@ -103,26 +104,204 @@ def delete_template(template_id):
+            content = content.replace(placeholder_with_spaces, str_value)
+            
+            # Handle format: {{variable_name}} (without spaces - for backward compatibility)
+            placeholder_no_spaces = f"{{{{{var_name}}}}}"
+            content = content.replace(placeholder_no_spaces, str_value)
+        
+        # Handle signature placeholders
+        if contract_signatures:
+            content = ContractTemplateService._replace_signature_placeholders(content, contract_signatures)
+        
+        # Log any unreplaced variables for debugging
+        import re
```

Ruff (0.13.3)

Redefinition of unused `re` from line 4

Remove definition: `re`

(F811)